plugins/modules/ndo_l3out_template.py

```diff
@@ -0,0 +1,936 @@
+seealso:
+- module: cisco.mso.mso_schema_template_vrf
+- module: cisco.mso.ndo_template
+- module: cisco.mso.ndo_l3_domain
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a new L3Out object without routing protocols
+  cisco.mso.ndo_l3out_template:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    l3out_template: l3out_template
+    name: "l3out_1"
+    vrf:
+      name: "VRF1"
+      schema: "Schema1"
+      template: "Template1"
+    state: "present"
+
+- name: Query a L3Out object with name
+  cisco.mso.ndo_l3out_template:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    l3out_template: l3out_template
+    name: "l3out_1"
+    state: "query"
+  register: query_l3out_name
+
+- name: Update a L3Out object with UUID
+  cisco.mso.ndo_l3out_template:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    l3out_template: l3out_template
+    uuid: "{{ query_l3out_name.current.uuid }}"
+    name: "l3out_1_updated"
+    description: "updated description"
+    state: "present"
+
+- name: Create a new L3Out object with routing protocols
+  cisco.mso.ndo_l3out_template:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    l3out_template: l3out_template
+    name: "l3out_1_new"
+    vrf:
+      name: "VRF1"
+      schema: "Schema1"
+      template: "Template1"
+    routing_protocols: ["bgp", "ospf"]
+    bgp:
+      inbound_route_map: "ans_route_map"
+      outbound_route_map: "ans_route_map"
+      route_dampening_ipv4: "ans_route_map"
+      route_dampening_ipv6: "ans_route_map"
+    ospf:
+      area_id: "0.0.0.1"
+      area_type: "regular"
+      cost: 1
+      send_redistributed_lsas: true
+      originate_summary_lsa: true
+      suppress_forwarding_addr_translated_lsa: true
+      originate_default_route: "only"
+      originate_default_route_always: false
+    interleak: "ans_route_map"
+    static_route_redistribution: "ans_route_map"
+    connected_route_redistribution: "ans_route_map"
+    attached_host_route_redistribution: "ans_route_map"
+    state: "present"
+
+- name: Clear bgp and ospf routing protocol from the existing L3Out
+  cisco.mso.ndo_l3out_template:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    l3out_template: l3out_template
+    name: "l3out_1_new"
+    vrf:
+      name: "VRF1"
+      schema: "Schema1"
+      template: "Template1"
+    routing_protocol: ["bgp"]
+    bgp:
+      inbound_route_map: ""
+      outbound_route_map: ""
+      route_dampening_ipv4: ""
+      route_dampening_ipv6: ""
+    ospf: {}
+    state: "present"
+
+- name: Query a L3Out object with name
+  cisco.mso.ndo_l3out_template:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    l3out_template: l3out_template
+    name: "l3out_1"
+    state: "query"
+  register: query_l3out_name
+
+- name: Query a L3Out object with UUID
```

Comment:
so this has been decided it should be standard functionality? Should we raise issue to track this and backport it to the other existing modules that only did uuid for the name change?

Reply:
It is still in discussion.